src/gateway/hal/mappers_classic/thermostat.py

```diff
@@ -48,12 +48,12 @@ def orm_to_dto(orm_object):  # type: (EepromModel) -> ThermostatDTO
                                                   start_day_2=data[field][4],
                                                   end_day_2=data[field][5],
                                                   temp_day_2=data[field][6])
-        return ThermostatDTO(id=data['id'],
+        return ThermostatDTO(number=data['number'],
```

Comment:
The data here comes from the gen2 eeprom configuration, etc. ThermostatConfiguration which still uses an `id` field to represent the number slots used by the firmware.

```suggestion
        return ThermostatDTO(number=data['id'],
```